app/bundles/LeadBundle/Validator/Constraints/SegmentDateValidator.php

```diff
@@ -28,11 +28,20 @@ public function validate($filters, Constraint $constraint): void
                     continue;
                 }
 
-                $dateString = str_replace('%', '', $parameterValue);
+                if (str_contains($parameterValue, '%')) {
+                    return;
+                }
```

Comment:
Why was this changed? Can you add some value containing the `%` character to your test? Perhaps I don't get why was it replaced with an empty string in the previous version of the code.

Reply:
For contains/startwith etc strin value is like `%2023-01-01`. These formats I cannot validated with \DateTime::createFromFormat

But anyway I marked it as WIP still need some things to do

Reply:
I've added more commits.
We tried to validate some mistakes. 
For example, empty/not empty or regexp we have to skip. Also values for like not like expressions are skipped.
Now should cover all possible values to validate